docs/notifications/03-showrunners-scaffold/03-Examples/08-Showrunner-Example-Btc-Tracker.mdx

```diff
@@ -0,0 +1,757 @@
+        msg: payloadMsg, // Internal Message
+        recipient: subscriberObj.subscriber, // Recipient
+      };
+
+      // Send notification
+      this.sendNotification({
+        recipient: payload.recipient, // new
+        title: payload.notifTitle,
+        message: payload.notifMsg,
+        payloadTitle: payload.title,
+        payloadMsg: payload.msg,
+        notificationType: payload.type,
+        simulate: simulate,
+        image: null,
+      });
+    } catch (error) {
+      this.logError(`Error sending notification: ${error}`);
+    }
+  }
+}
+```
+
+🤯Those were a lots of code out there. Let's understand what is actually happening there and what conditions trigger the notifications in different cases.
+
+- **Case 1: Both percent change and time interval is enabled** - When a user opts in to both these settings, what the user want is to receive a notification for their selected tokens when there is a particular change in price and it occured within the time interval.
+  So, the basic logic here is:
+
+```jsx
+if (userDBValue.lastCycle + settingUserValue2 == CYCLES) {
+  if (changePercentage >= settingUserValue1) {
+  }
+}
+```
+
+We just fetched the prices from the CMC API and using the previous price stored in database as per user, we can calculate the `changePercentage` value.
+For the `CYCLES` variable, everytime our showrunners framework is executed it is incremented by 3 as the lowest `ticker` value in the slider is 3. You can change it as per your channel and logic. This helps us to calculate when a new user will receive a notification based on on which cycle did he opted in.
+Also, there are 3 conditions that you need to lookout for:
+
+- i) What happens when time is triggered but not percentage?
+- ii) What happens if a user opts-in, opts-out and then again after several days opt-in?
+- iii) What happens if someone changes their time-interval settings?
+
+We have already handled these edge cases in the code. Test yourself and see if you can find them😉.
+
+- **Case 2: Only percent change is enabled** - Here, a user want to receive notification when there is a particular change in price. So, the basic logic here is:
+
+```js
+// Condition to trigger notification
+if (Number(changePercentage) >= userValue) {
+}
+```
+
+The calculation for the `changePercentage` is same like `Case 1`. The only difference here is we use the `globalChangePercentage` instead of the users last price.
+
+- **Case 3: Only time interval is enabled** - Here, a user want to receive notification as per their chosen interval. So, the basic logic here is:
+
+```js
+// Condition to trigger notification
+if (userDBValue.lastCycle + userValue == CYCLES) {
+}
+```
+
+The calculation and significance of the `CYCLES` variable is explained in `Case 1`.
+
+This wraps up the channel logic. Now, let's move onto buidling the cron-jobs file and model file.
+
+### Step 8: Create a `btcTickerModel.ts` file in the folder.
+
```

Comment:
can you explain in this step how settingsValue will solve the three edge-cases you mentioned above, thanks.